### DISCUSSION TITLE
How can you preview the formatting syntax for your file in your repository?

### DISCUSSION THREAD

**Discussion by akim67212** [community/community#138518] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Question

### Body

Lets say you created a new file in your repository, and you have written and formatted it using the syntax. How are you able to preview the new file in order to make sure it looks good???

**Comment by mahmoud-abu-sultan** [comment]
To preview a newly created and formatted file in GitHub, follow these steps:

Commit and Push the File: First, make sure that your new file is committed and pushed to the repository. GitHub only allows file preview for files that are already part of the repository.

File Preview in GitHub:

Navigate to your repository on GitHub.
Locate the newly created file in the file list.
Click on the file name to open it.
GitHub will automatically render and preview certain file formats (e.g., Markdown .md, text files .txt, and other supported formats) in a readable format.
Markdown Preview (if you're working with .md files): If you are using Markdown for documentation, GitHub automatically renders the formatted output of Markdown files, so you can see headings, lists, links, code blocks, etc., as they would appear on a webpage.

Raw File View: If you want to view the file's raw content (e.g., to ensure syntax accuracy in other formats like code), you can click on the Raw button at the top-right corner of the file view.
#MAHMOUD.T
